src/lib/es2015.proxy.d.ts

```diff
@@ -1,21 +1,23 @@
-interface ProxyHandler<T extends object> {
+interface ProxyHandler<T extends object, U extends object> {
     apply?(target: T, thisArg: any, argArray: any[]): any;
     construct?(target: T, argArray: any[], newTarget: Function): object;
     defineProperty?(target: T, p: string | symbol, attributes: PropertyDescriptor): boolean;
     deleteProperty?(target: T, p: string | symbol): boolean;
-    get?(target: T, p: string | symbol, receiver: any): any;
+    get?<K extends keyof U>(target: T, p: K, receiver: U): U[K];

```

Comment:
can you add a test for the program in the original bug? (https://github.com/microsoft/TypeScript/issues/20846)
I don't understand how inference of `U` would work to fix the bug for that program, since in that case there is no `receiver`.